packages/video_player/ios/Classes/VideoPlayerPlugin.m

```diff
@@ -264,16 +271,19 @@ + (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar>*)registrar {
                                   binaryMessenger:[registrar messenger]];
   FLTVideoPlayerPlugin* instance =
       [[FLTVideoPlayerPlugin alloc] initWithRegistry:[registrar textures]
-                                           messenger:[registrar messenger]];
+                                           messenger:[registrar messenger]
+                                           registrar:registrar];
   [registrar addMethodCallDelegate:instance channel:channel];
 }
 
 - (instancetype)initWithRegistry:(NSObject<FlutterTextureRegistry>*)registry
-                       messenger:(NSObject<FlutterBinaryMessenger>*)messenger {
+                       messenger:(NSObject<FlutterBinaryMessenger>*)messenger
+                       registrar:(NSObject<FlutterPluginRegistrar>*)registrar {
```

Comment:
Maybe reduce parameters to just the `registrar`, then access the `FlutterBinaryMessenger` and the `FlutterTextureRegistry` from there.